stdlib/public/core/Arrays.swift.gyb

```diff
@@ -26,7 +26,19 @@
 
 /// This type is used as a result of the _checkSubscript call to associate the
 /// call with the array access call it guards.
-public struct _DependenceToken {}
+///
+/// When the optimizer performs optimizations based on @_semantic calls, it
+/// sometimes wants to find calls that are semantically belonging together.
+/// For example, in an array subscript operation, it wants to find the
+/// _checkSubscript call that logically goes with the _getElement call.
+/// To do this we assign a return value from the _checkSubscript call to an
+/// argument of the _getElement call.
+///   let token = _checkSubscript(index, ... )
+///   return _getElement(index, ... , matchingSubscriptCheck: token)
+///
+/// This is public because it is used by the public function _checkSubscript.
```

Comment:
Making `_DependenceToken` @_versioned along with _checkSubscript  leads to massive slowdowns in resilient mode, because now the initializer of _DependenceToken cannot be declared @_inlineable (the compiler complains that this type is not @_fixed_layout and the init has to delegate to another initializer). As a result, creation of `_DependenceToken` instances is very expensive.